server/models/CardAttachementsPlugin.js

```diff
@@ -0,0 +1,138 @@
+const { Schema, model } = require("mongoose");
+const BasePlugin = require("./BasePlugin");
+const { v2: cloudinary } = require("cloudinary");
+
+const CardAttachementsSchema = new Schema({
+  imageUrl: {
+    type: String,
+    required: true
+  },
+  imagePublicId: {
+    type: String,
+    required: true
+  }
+}, { timestamps: true });
+
+const CardAttachementsPluginSchema = BasePlugin.discriminator("CardAttachementsPlugin", new Schema({
+  attachements: [{
+    type: Schema.Types.ObjectId,
+    ref: "CardAttachements"
+  }]
+}, { timestamps: true }));
+
+CardAttachementsPluginSchema.methods.get = async function ({ cardAttachementsPluginId }) {
+  if (!cardAttachementsPluginId) {
+    const err = new Error("Not a valid request!");
+    err.statusCode = 400;
+    throw err;
+  }
+  const data = await this.model("CardAttachementsPlugin").findById(cardAttachementsPluginId).populate("attachements");
+  return {
+    status: 200,
+    data
+  };
+};

```

Comment:
So, verify plugin middleware will already know the model exists. Are we able to just work off the `this` and update the data or do we have to do the query. Can you verify this for me? 

Also the arguments passed by the plugin handler won't give you the `cardAttachmentPluginId` it will be `{ body, params, query }`

Reply:
Ditto everywhere else.

Reply:
![Screenshot 11-17-2021 2-46-50 AM](https://user-images.githubusercontent.com/70457144/142117675-86f176ad-ce07-4906-8dbe-6b356d42bcb1.png)
using function declaration, the this method binds the method to the document and will be able to work. but however it doesn't work for arrow functions. So I made sure I don't use arrow functions :)